### PR TITLE
feat: add publish to cargo step on release

### PR DESCRIPTION
**Pull request by louisjoecodes**
- added workflow_dispatch so that we can manually trigger the publish step
- Secrets stored.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ on:
   push:
     tags:
       - "v*.*.*"
+  workflow_dispatch:
```
Comment (louisjoecodes) [inline .github/workflows/release.yml:5]: Adding this so that we can manually trigger the release workflow

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -145,3 +145,24 @@ jobs:
...
+  publish-to-cargo:
+    name: Publish to Cargo
+    runs-on: ubuntu-latest
+    needs: [release, check-versions-match]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
```
Comment (benfdking) [inline .github/workflows/release.yml]: I would remove this override as we don't use stable in sqruff. 

```suggestion
```

**Review by benfdking — CHANGES_REQUESTED** [review]
Small change on toolchain

**Review by benfdking — APPROVED** [review]
Send it!

**Pull request merged**